sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2023-2024.bpmn

```diff
@@ -608,117 +639,169 @@
-      <bpmn:incoming>Flow_0n7won7</bpmn:incoming>
-      <bpmn:outgoing>Flow_10kp7wy</bpmn:outgoing>
+      <bpmn:incoming>Flow_1x86r10</bpmn:incoming>
+      <bpmn:outgoing>Flow_0jntyac</bpmn:outgoing>
     </bpmn:intermediateThrowEvent>
-    <bpmn:intermediateThrowEvent id="Event_1b81z89" name="calculatedDataPartnerStudentStudyWeeks">
+    <bpmn:intermediateThrowEvent id="Event_18vto57" name="Calculate Total Child Care Costs">
       <bpmn:extensionElements>
         <zeebe:ioMapping>
-          <zeebe:output source="=if partner1StudentStudyWeeks = null then 0 else partner1StudentStudyWeeks" target="calculatedDataPartnerStudentStudyWeeks" />
+          <zeebe:output source="=if calculatedDataChildCareCost &#62; 0&#10;  then min(((offeringCourseLoad/100) * dmnPartTimeProgramYearMaximums.limitWeeklyChildCare &#10;    * calculatedDataOfferingWeeksMultiplier * calculatedDataTotalEligibleDependentsForChildCare),&#10;  calculatedDataChildCareCost)&#10;else 0" target="calculatedDataTotalChildCareCost" />
         </zeebe:ioMapping>
       </bpmn:extensionElements>
-      <bpmn:incoming>Flow_1d1sl5k</bpmn:incoming>
-      <bpmn:outgoing>Flow_02m6u8j</bpmn:outgoing>
+      <bpmn:incoming>Flow_19cfg17</bpmn:incoming>
+      <bpmn:outgoing>Flow_1x86r10</bpmn:outgoing>
     </bpmn:intermediateThrowEvent>
-    <bpmn:intermediateThrowEvent id="Event_02st413" name="calculatedDataPartner1TotalIncome">
+    <bpmn:exclusiveGateway id="Gateway_0oywejd">
+      <bpmn:incoming>Flow_1mlb12p</bpmn:incoming>
+      <bpmn:outgoing>Flow_0x59561</bpmn:outgoing>
+      <bpmn:outgoing>Flow_1q9cwdm</bpmn:outgoing>
+      <bpmn:outgoing>Flow_1yysw0i</bpmn:outgoing>
+    </bpmn:exclusiveGateway>
+    <bpmn:intermediateThrowEvent id="Event_1fyviov" name="Calculate Total Transportation Allowance">
       <bpmn:extensionElements>
         <zeebe:ioMapping>
-          <zeebe:output source="=studentDataEstimatedSpouseIncome" target="calculatedDataPartner1TotalIncome" />
+          <zeebe:output source="=// If the student is receiving an additional ammount based on a clinical or practicum&#10;// placement then reduce transportation allowance limit accordingly.&#10;if (calculatedDataAdditionalTransportPlacement = true)&#10;then&#10;    dmnPartTimeProgramYearMaximums.limitAdditionalTransportationAllowance - dmnPartTimeProgramYearMaximums.limitAdditionalTransportationPlacement&#10;  else&#10;    dmnPartTimeProgramYearMaximums.limitAdditionalTransportationAllowance" target="calculatedDataAdditionalTransportEstimatedCost" />
+          <zeebe:output source="=min(calculatedDataAdditionalTransportCost,&#10;calculatedDataAdditionalTransportEstimatedCost)" target="calculatedDataAdditionalWeeklyTransportationCost" />
+          <zeebe:output source="=calculatedDataAdditionalWeeklyTransportationCost * calculatedDataAdditionalTransportWeeks * dmnPartTimeProgramYearMaximums.limitAdditionalTransportationKMReduction" target="calculatedDataAdditionalTransportationMax" />
```

Comment:
Is it right to assume that the only difference between the calculations for the 4 variables in the top 2 conditions below is one extra multiplication?

![image](https://github.com/bcgov/SIMS/assets/61259237/62f37880-5433-43b5-b172-f0f9e8b45a8a)


```
calculatedDataAdditionalWeeklyTransportationCost * calculatedDataAdditionalTransportWeeks * dmnPartTimeProgramYearMaximums.limitAdditionalTransportationKMReduction
```

If that is true the logic for the 4 variables can be centralized if a new variable is created to keep the multiplier as dmnPartTimeProgramYearMaximums.limitAdditionalTransportationKMReduction or 1, which means that an alternative gateway can be created.
Please let me know if the assumption is right. If it is, I can provide the workflow change if needed.

Reply:
Yes, that is indeed the case in calculating `calculatedDataAdditionalTransportationMax`. Please feel free to provide an alternative workflow change for carrying out optimal calculations.

Reply:
The workflow can be changed as below where the highlighted calculations would determine the value for a new variable `limitAdditionalTransportationKMReduction`.
For the top flow (item 1), its values would be 1.
For the middle flow (item 2), its value would be `dmnPartTimeProgramYearMaximums.limitAdditionalTransportationKMReduction`

Then the `calculatedDataAdditionalTransportationMax` would be calculated in the item "3" only once alongside with the other 3 variables.
```
calculatedDataAdditionalWeeklyTransportationCost * calculatedDataAdditionalTransportWeeks * limitAdditionalTransportationKMReduction
```

![image](https://github.com/bcgov/SIMS/assets/61259237/2fb6019e-f00e-469c-9926-b00e0168bbd9)

Please let me know if its make sense.

Reply:
Thanks @andrewsignori-aot. This alternative workflow is great. I extended this idea and put `limitAdditionalTransportationKMReduction` (renamed to `calculatedDataAdditionalTransportKMReduction`) into the event prior to the gateway. Instead of 3 routes, the gateway is now using 2 routes with km = 0 and km > 0, and the workflow is very concise and inclusive in considering all scenarios used in e2e tests. Please have a look.

![Screenshot 2024-02-21 104252](https://github.com/bcgov/SIMS/assets/148148914/c9138c82-029c-4bcb-be70-e04a6897c552)

Reply:
Thanks for doing the changs 👍